lib.php

```diff
@@ -4,30 +4,25 @@
 defined('MOODLE_INTERNAL') || die();
 class enrol_payment_plugin extends enrol_plugin {
 
-    public function enrol_page_hook(stdClass $instance){  
-        global $CFG, $USER, $OUTPUT, $PAGE, $DB;
+     public function enrol_page_hook(stdClass $instance){  
+         global $CFG, $USER, $OUTPUT, $PAGE, $DB;
 
-        //saves everything about to happen
         ob_start();
-        echo '<p> this works <p>';
+        
+        echo "<p><b>To enrol into a class please pay the amount below</b></p>";
 
-        //Check the price of the course 
-        if (abs($cost) < 0.01) { 
-            //returns "This course is free."
-            echo '<p>'.get_string('nocost', 'enrol_payment').'</p>';
-        }
+        echo "<p>Price: $</p>";
 
+        if(array_key_exists('paymentButton',$_POST)){
```

Comment:
Dont access POST directly, big security issue. Use optional_param or required_param. They can check POST as well. They validate it for safety, and strip out nasties